src/Language/Haskell/Brittany/Internal/LayouterBasics.hs

```diff
@@ -299,7 +299,7 @@ filterAnns ast =
 -- b) after (in source code order) the node.
 hasAnyCommentsBelow :: Data ast => GHC.Located ast -> ToBriDocM Bool
 hasAnyCommentsBelow ast@(L l _) =
-  List.any (\(c, _) -> ExactPrint.commentIdentifier c > l)
+  List.any (\(c, _) -> ExactPrint.commentIdentifier c > ExactPrint.Utils.rs l)
```

Comment:
This function returns a "bad" real source span when a "good" one isn't available. Perhaps that messes up sorting somehow? Note that regular source spans can't be compared. 

https://hackage.haskell.org/package/ghc-exactprint-1.2.0/docs/Language-Haskell-GHC-ExactPrint-Utils.html#v:rs